plugins/scaffolder-backend/src/scaffolder/tasks/logger.ts

```diff
@@ -107,20 +108,16 @@ export class WinstonLogger implements RootLoggerService {
 
     let redactionPattern: RegExp | undefined = undefined;
 
-    const replace = (obj: TransformableInfo) => {
-      for (const key in obj) {
-        if (obj.hasOwnProperty(key)) {
-          if (typeof obj[key] === 'object') {
-            obj[key] = replace(obj[key] as TransformableInfo);
-          } else if (typeof obj[key] === 'string') {
-            obj[key] = obj[key]?.replace(redactionPattern, '[REDACTED]');
-          }
-        }
-      }
-      return obj;
-    };
     return {
-      format: format(replace)(),
+      format: format((obj: TransformableInfo) => {
+        if (!redactionPattern || !obj) {
+          return obj;
```

Comment:
Wouldn't it be better to use here `undefined`? 
Though it's kinda the same, it's easier to read the code faster

Reply:
You mean as the return value? We'd want to return obj if we don't have a redaction pattern